internal/transform/suites.go

```diff
@@ -0,0 +1,108 @@
+package transform
+
+import (
+	"context"
+
+	"github.com/joshdk/go-junit"
+	"github.com/mdelapenya/junit2otlp/internal/config"
+	"github.com/mdelapenya/junit2otlp/internal/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type OtelProvider interface {
+	Tracer(string) trace.Tracer
+	Meter(string) metric.Meter
+}
+
+func TransformAndLoadSuites(ctx context.Context, cfg *config.Config, provider OtelProvider,
```

Comment:
nit: is the function name duplicating the package name? Having to use it like this `transform.TransformAndLoadSuites` seems weird. The natural `AndLoadSuites` sounds incorrect, although not sure about `WithLoadSuites`. wdyt?

Reply:
I will think about this on my walk and adjust the name when I get back.